### ISSUE TITLE
DCEVM doesn't work as expected with jbr_dcevm-17-osx-x64-b135.1.tar.gz

### ISSUE THREAD

**Issue by NicklasWallgren** [JetBrains/JetBrainsRuntime#96]
I have compared `jbr_dcevm-11_0_13-osx-x64-b1751.16.tar.gz` with `jbr_dcevm-17-osx-x64-b135.1.tar.gz` and I have noticed that the hot reload doesn't work as expected in dcevm-17, compared to the dcevm-11 version.

Re-compiled files only gets reloaded once in dcevm-17, and new changes isn't picked up.

Here you got a short screen recording which shows the issue.
https://drive.google.com/file/d/1kMsgmm3aa5UtUOCdOQEnxup2_zR69YDO/view?usp=sharing

Output from dcevm-17
```
-- Hotswap works, first time encountering TestController
[23.959s][debug][redefine,class,load] loading name=com.example.hexagonalarchitecture.adapter.driver.api.TestController kind=102 (avail_mem=27232K)
[23.991s][debug][redefine,class,load] Comparing different class versions of class com/example/hexagonalarchitecture/adapter/driver/api/TestController
[23.991s][debug][redefine,class,load] loaded name=com.example.hexagonalarchitecture.adapter.driver.api.TestController (avail_mem=27232K)
[23.991s][info ][redefine,class,load] redefined name=com.example.hexagonalarchitecture.adapter.driver.api.TestController, count=1 (avail_mem=27232K)
[24.122s][info ][redefine,class,timer] vm_op: all=163  prologue=33  doit=130
[24.122s][info ][redefine,class,timer] doit: heap iterate=55  fullgc=0
HOTSWAP AGENT: 17:10:34.829 WARNING (org.hotswap.agent.watch.nio.WatcherNIO2) - Watcher on /<REMOVED>/hexagonal-architecture/build/classes/java/main/com/example/hexagonalarchitecture/domain/service not valid, removing path=
---- No longer hotswaps on rebuild 
[58.342s][info ][redefine,class,timer] vm_op: all=92  prologue=3  doit=89
[58.342s][info ][redefine,class,timer] doit: heap iterate=50  fullgc=0
---- No longer hotswaps on rebuild 
[73.388s][info ][redefine,class,timer] vm_op: all=92  prologue=1  doit=91
[73.388s][info ][redefine,class,timer] doit: heap iterate=51  fullgc=0
---- No longer hotswaps on rebuild 
[103.926s][info ][redefine,class,timer] vm_op: all=217  prologue=31  doit=186
[103.926s][info ][redefine,class,timer] doit: heap iterate=88  fullgc=0
```

Output dcevm-11 recompiling the same file
```
[19.380s][debug][redefine,class,load] loading name=com.example.hexagonalarchitecture.adapter.driver.api.TestController kind=102 (avail_mem=21648K)
[19.403s][debug][redefine,class,load] Comparing different class versions of class com/example/hexagonalarchitecture/adapter/driver/api/TestController
[19.403s][debug][redefine,class,load] loaded name=com.example.hexagonalarchitecture.adapter.driver.api.TestController (avail_mem=21648K)
[19.406s][info ][redefine,class,load] redefined name=com.example.hexagonalarchitecture.adapter.driver.api.TestController, count=1 (avail_mem=21648K)
[19.569s][info ][redefine,class,timer] vm_op: all=194  prologue=32  doit=162
[19.569s][info ][redefine,class,timer] doit: heap iterate=79  fullgc=0
-- Works
[30.204s][debug][redefine,class,load ] loading name=com.example.hexagonalarchitecture.adapter.driver.api.TestController kind=102 (avail_mem=32652K)
[30.229s][debug][redefine,class,load ] Comparing different class versions of class com/example/hexagonalarchitecture/adapter/driver/api/TestController
[30.229s][debug][redefine,class,load ] loaded name=com.example.hexagonalarchitecture.adapter.driver.api.TestController (avail_mem=32652K)
[30.229s][info ][redefine,class,load ] redefined name=com.example.hexagonalarchitecture.adapter.driver.api.TestController, count=2 (avail_mem=32652K)
[30.348s][info ][redefine,class,timer] vm_op: all=146  prologue=27  doit=119
[30.348s][info ][redefine,class,timer] doit: heap iterate=67  fullgc=0
-- Works
[52.235s][debug][redefine,class,load ] loading name=com.example.hexagonalarchitecture.adapter.driver.api.TestController kind=102 (avail_mem=43708K)
[52.262s][debug][redefine,class,load ] Comparing different class versions of class com/example/hexagonalarchitecture/adapter/driver/api/TestController
[52.262s][debug][redefine,class,load ] loaded name=com.example.hexagonalarchitecture.adapter.driver.api.TestController (avail_mem=43708K)
[52.262s][info ][redefine,class,load ] redefined name=com.example.hexagonalarchitecture.adapter.driver.api.TestController, count=3 (avail_mem=43708K)
[52.383s][info ][redefine,class,timer] vm_op: all=150  prologue=30  doit=120
[52.383s][info ][redefine,class,timer] doit: heap iterate=60  fullgc=0
```

Linked issue for reference, https://github.com/HotswapProjects/HotswapAgent/issues/405

**Comment by NicklasWallgren** [comment]
The default `jbr-17-osx-x64-b135.1` seems to work for this test purpose, the changes are hot reloaded for each re-compilation.

```
152.044s][debug][redefine,class,load        ] loaded name=com.example.hexagonalarchitecture.adapter.driver.api.TestController (avail_mem=57500K)
[152.049s][info ][redefine,class,load        ] redefined name=com.example.hexagonalarchitecture.adapter.driver.api.TestController, count=6 (avail_mem=57500K)
[152.053s][debug][redefine,class,nmethod     ] Marked 0 dependent nmethods for deopt
[152.061s][info ][redefine,class,timer       ] vm_op: all=0  prologue=0  doit=0
[152.061s][info ][redefine,class,timer       ] redefine_single_class: phase1=0  phase2=0
```

**Comment by NicklasWallgren** [comment]
Should be fixed as of https://github.com/JetBrains/JetBrainsRuntime/releases/tag/jbr17_0_1b164.8. 

I'll verify and then close this issue.

**Comment by NicklasWallgren** [comment]
Fixed as of `JBR-17.0.1.12-164.8-dcevm`. Thanks for the help @skybber.
